### DISCUSSION TITLE
Incorrectly recovered and inconsistent data after cluster recovery

### DISCUSSION THREAD

**Discussion by krivulcik** [ravendb/ravendb#14448] [Q&A]
This is all with 5.3.103, both server and studio.

We had one node lose data this weekend. Now, after the recovery, the node seems to have incorrectly recovered data: There are documents with outdated data, and also some documents are missing.

The replication seems to have been completed, and I'm unsure how to investigate further.

The data disk lost the data. After the loss, we formatted the data disk and started the RavenDB service. The data was replicated, and the node was indexing the documents.

However, the replicated database doesn't contain some documents, and some documents seem to be outdated: We have at least one document not present in the target database and at least one of the documents is not the latest version.

The offending document has revisions. We can't see the revisions on a source node (B and C are healthy). The revisions are available on the target node A, but there are only three revisions, and all the data in the revisions is different from what should be available. 

In effect, it seems that the replication didn't finish successfully.

The database etags are almost the same on all nodes (not exactly the same, but the values within around 10 of each other), from which I deduce that the replication is considered to be completed.

We have moved the database directory out of the way to attempt a whole-node recovery from scratch.
However, it impacted our production environment, so I have brought the node down for the time being.
I'll retry again later, outside working hours.

Creating a debug package took a very long time, again, degrading our service.
I'll retry creating it again, and I can provide it if that is useful for investigation.

Also, the remaining nodes (B and C) have a very different view of the database state.

Node B:
![image](https://user-images.githubusercontent.com/7997055/175963559-de6c59c2-4b5d-454b-b531-1975036fb52c.png)

Node C:
![image](https://user-images.githubusercontent.com/7997055/175963584-4b685720-9b48-43ec-b307-f8f5149b1ab1.png)

Although node A is down, node C thinks it is available and fully working member.
Although node C is up, node B thinks it is unavailable.

Also, maybe unrelated to the issue at hand, when I open the page /studio/index.html#databases/manageDatabaseGroup?&database=data-pep, the websocket connection doesn't work, and the browser fires around 20 requests per second to the /databases/.../tasks and /databases endpoints.

Is it possible to investigate in more detail here to prevent such a situation from happening in the future?

**Comment by garayx** [comment]
Is it possible that you had writes/reads to the database that was in rehab state (the one which was getting recovered) ? I guess this can bring to data missmatch from other nodes.
